File: src/main/java/com/mobilesolutionworks/android/viper/VIPERPresenter.java
package com.mobilesolutionworks.android.viper;

import com.mobilesolutionworks.android.app.controller.WorksController;

/**
 * Created by yunarta on 5/5/17.
 */

public class VIPERPresenter<V extends VIPERView> extends WorksController {

    private V view;

    public void setView(V view) {
        this.view = view;
        onAttached();
    }

    protected V view() {
        return view;
    }

    protected void onAttached() {

    }
}

File: src/main/kotlin/com/mobilesolutionworks/android/viper/VIPERPresenter.kt
package com.mobilesolutionworks.android.viper

import com.mobilesolutionworks.android.app.controller.WorksController

/**
 * Created by yunarta on 14/5/17.
 */

open class VIPERPresenter<V : VIPERView> : WorksController() {

    private var view: V? = null

    fun setView(view: V) {
        this.view = view
        onAttached()
    }

    protected fun view(): V? {
        return view
    }

    open protected fun onAttached() {

    }
}